src/backlight/strategies/exit.py

```diff
@@ -36,6 +36,45 @@ def _exit_transaction(
     return Transaction(timestamp=exit_index, amount=-trade.sum())
 
 
+def _exit_transaction_gain_loss(
+    df: pd.DataFrame,
+    trade: pd.Series,
+    loss_threshold: float,  # Negative
+    gain_threshold: float,  # Positive
+) -> Transaction:
+
+    # Get an exit indice that satisfies gain or loss limit
+    current_signal = df["pred"][0]
+    if df["pred"][0] == 1.0:  # If signal UP, we buy at ask, sell at bid
```

Comment:
For the first step, how about just using `mid` price? In that case, we can write logic simply. Or, we can write unittest, otherwise it is quite difficult to judge the algorithm is fine.

Reply:
When using `mid`, logic should be:

```
def exit_at_loss_and_gain(
    mkt: MarketData,
    entries: Trades,
    max_holding_time: pd.Timedelta,
    loss_threshold: float,
    gain_threshold: float,
) :
    assert loss_threshold >= 0.0
    assert gain_threshold >= 0.0

    def _exit_at_loss_and_gain(df: pd.DataFrame, trade: pd.Series) -> pd.Series:
        prices = df.mid

        amount = trade.sum()
        entry_price = prices.iloc[0]
        pl_per_amount = np.sign(amount) * (prices - entry_price)
        is_stop_loss = pl_per_amount <= -initial_stop
        is_take_gain = pl_per_amount >= gain_threshold

        return is_initial_stop | is_take_gain

    return exit_by_max_holding_time(mkt, None, entries, max_holding_time, _exit_at_loss_and_gain)
```

Reply:
If we want to ask/bid pricing for loss cut, 
I think it is better to create functions that will replace `df.mid`.

```
prices = calculate_price(mkt, sig)

def calculate_price(mkt, sig):
    if sig == UP:
        return mkt.ask
    if sig == DOWN:
        return mkt.bid
    return mkt.mid
```

Reply:
Ok I got it. I just updated it to take mid price for now.